### PR TITLE
Core/BG fix .character changefaction pvpstats victories amount

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
Bug: with the current system, after use of `.character changefaction` command the victories and looses of the player will be swapped.

This PR solves the issue by binding the victories to the players in addition of binding them to the BGs.

### CODE REVIEW

File: src/server/game/Battlegrounds/Battleground.cpp [hunk 1]
```diff
@@ -784,19 +784,22 @@ void Battleground::EndBattleground(uint32 winner)
...
-            stmt->setUInt32(4, score->second->GetHonorableKills());
-            stmt->setUInt32(5, score->second->GetBonusHonor());
-            stmt->setUInt32(6, score->second->GetDamageDone());
-            stmt->setUInt32(7, score->second->GetHealingDone());
-            stmt->setUInt32(8, score->second->GetAttr1());
-            stmt->setUInt32(9, score->second->GetAttr2());
-            stmt->setUInt32(10, score->second->GetAttr3());
-            stmt->setUInt32(11, score->second->GetAttr4());
-            stmt->setUInt32(12, score->second->GetAttr5());
+            bool isWinner = team == winner ? true : false;
```
Comment (Treeston) [inline src/server/game/Battlegrounds/Battleground.cpp]: equivalent to `bool isWinner = (team == winner)`

File: src/server/game/Battlegrounds/Battleground.cpp [hunk 2]
```diff
@@ -784,19 +784,22 @@ void Battleground::EndBattleground(uint32 winner)
...
+
+            stmt->setUInt32(0,  battlegroundId);
+            stmt->setUInt32(1,  player->GetGUID().GetCounter());
+            stmt->setBool  (2,  isWinner);
```
Comment (Treeston) [inline src/server/game/Battlegrounds/Battleground.cpp]: Could simply be `stmt->setBool(2, team == winner)`?

File: sql/base/characters_database.sql [hunk 3]
```diff
@@ -2510,7 +2510,7 @@ CREATE TABLE `updates` (
```
Comment (pete318) [inline sql/base/characters_database.sql]: The extra winner column should also be added to the create statement in here too.

**Comment by ShinDarth** [comment]
@Treeston @pete318 done, thanks

**Pull request merged**